src/main/scala/io/iohk/ethereum/network/PeerManagerActor.scala

```diff
@@ -71,6 +70,15 @@ class PeerManagerActor(
   // Subscribe to the handshake event of any peer
   peerEventBus ! Subscribe(SubscriptionClassifier.PeerHandshaked)
 
+  // Subscribe to messages received from peers to maintain stats.
+  peerEventBus ! Subscribe(MessageSubscriptionClassifier)
```

Comment:
My initial tought is that we should probably have separate actor that would gather those stats, to not interfere with `PeerManagerActor` other tasks. (there can be a lot of messages from a lot of peers) my plan would be:
- to move publication of the `PeerHandshaked` event to `PeerManagerActor`. `PeerActor` is child of `PeerManager` so it can inform `PeerManager` via `context.parent`
- then `PeerManagerActor` would be sole publisher of `PeerHandshaked` and `PeerDisconnected` events.
- then create actor `PeerStatiticsActor` which would subscirbe to those two events + all messages. This actor would maintain stats for all handshaked connections.
- `PeerStatiticsActor` could respons two different messages `getStatsForAll` and `getStatsForPeer(peerId)`

Having `PeerHandshaked` and `PeerDisconneted` events published from the same actor is important as this guarantees that `PeerEventBus` always receive `PeerHandshaked` first and `PeerDisconneted` second.

Reply:
Hm, yes, although it sounds a bit more complicated change, and also make pruning more complicated since now it would involve an extra message exchange. Do you want all this in this PR?

Reply:
I think this change in tracking should be done in this pr, as in case of many peers i would really like to avoid overloading PeerManagerActor with all those messages.

Also I think that this pr should be dealt with first and only after merging this one we should focus on merging pruning connections. (as i really do not like idea of dropping random incoming connections without any stats)

Reply:
Yeah that was the plan, to merge this and update the other.

Reply:
Moving the publishing of `PeerHandshakeSuccessful` from `PeerActor` to `PeerManagerActor` could result in subscribers seeing a `MessageFromPeer` first and a the successful handshake later, because the `PeerActor` keeps publishing messages to the even bust, while it has no control over when its parent will publish the handshake on its behalf. 

How about just introducing the `PeerStatisticsActor` in this PR and not try to modify who publishes handshakes and disconnects? It seems unrelated to this PR since I only gather messages, which only ever get published from handshaked peers. It's not a problem if the stats actor misses a disconnection event because eventually the statistics will be overwritten.

Reply:
Done, the statistics are gathered in the `PeerStatisticsActor`.